### DISCUSSION TITLE
automatically run pinned programs

### DISCUSSION THREAD

**Discussion by spetauskas** [HerMajestyDrMona/Windows11DragAndDropToTaskbarFix#37] [Ideas]
hi, 
I think it would be nice feature if this program could open pinned programs on taskbar automatically. let's say if i drag file on pinned program, which is not running, i can wait for longer time (maybe 3-6 sec (in configure file everyone could set time for they need)) program or file explorer launches automatically and the i can continue dragging file to opened program/explorer. another way would be if I drag file on pinned program and while hovering on icon I press right mouse button, it runs program/explorer.

**Comment by HerMajestyDrMona** [comment]
Interesting, I thought that it's already possible in the program, but now I see that it is not.
For now, you could use `UseTheNewBestMethodEver=0` to use the old method that opens every pinned app, however it's limited to the first 10 taskbar icons :(

I'll try to remember to add this feature in the next version.

**Reply by spetauskas** [reply]
now it's working if changed UseTheNewBestMethodEver=0 from 1 to 0. but I would like see more control of starting program. bigger time delay or extra action (like presing mouse button) :) sorry if I making funny request, I'm not programer, so this is like magic for me :D
By the way it's super huge help for  making this program 👍 I was considering to go threw all reinstalling windows 10 nonsense because of this new anti feature :D

**Reply by HerMajestyDrMona** [reply]
Glad that you like it :) Don't worry, I'll add this thing in the next version, so all features of the `UseTheNewBestMethodEver` will be available.

**Comment by HerMajestyDrMona** [comment]
Hello again.

Sorry for the delay. This feature has been added in [ver. 2.1.0.0](https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix/blob/main/CHANGELOG.md#ver-2100----2021-12-13-1600-cet).

You can enable it by the following configuration option:
```
AutoOpenPinnedAppsEvenWhenNoWindowActive=1
```

You can also tweak the delay after opening the program in order to minimize issues related to it:
```
HowLongSleepAfterOpeningPinnedAppMilliseconds=500
```

Greetings.
